src/lintsampler/lintsampler.py

```diff
@@ -0,0 +1,526 @@
+            The shape of each distinct grid
+        ngridentries : list of integers, optional, used if eval_type == 'freesample'
+            The number of entries in the 
+        nedgegridentries : list of integers, optional, used if eval_type == 'freesample'
+            The number of entries in the offset grids
+        x0 : list of numpy arrays, optional, used if eval_type == 'freesample'
+            The array of the first vertex of the grids
+        x1 : list of numpy arrays, optional, used if eval_type == 'freesample'
+            The array of the last vertex of the grids
+
+        
+        """
+
+        # check the validity of the input
+        if len(cells)==0:
+            raise ValueError("LintSampler._setgrid: you must specify an evaluation domain with a tuple of arrays or a list of tuples of arrays. See documentation for details.")
+            # or, we could here drop into an adaptive grid selection
+            # or, we could sample on a unit hypercube with dimensions of the pdf
+
+        # 0. are we in the 1d case? the input is just a single tuple or 1d array.
+        # i.e. cells = np.linspace(-12,12,100)
+        if (isinstance(cells[0],float) | (isinstance(cells[0],int))): # do we need the integer option here? I think no?
+            self.eval_type='gridsample'
+
+            # override the inferred dimensionality
+            self.dim = 1
+
+            # set the arrays
+            # we are being passed the array directly
+            self.edgearrays = cells
+            self.edgedims = len(cells)
+
+
+        # 1. tuples defining the array -> make arrays, pass to gridsample
+        # i.e. cells = (np.linspace(-12,12,100),np.linspace(-4,4,50))
+        if isinstance(cells,tuple):
+
+            if isinstance(cells[0],tuple):
+                raise ValueError("LintSampler: Cells must be a single tuple or a list of tuples (i.e. not a tuple of tuples).")
+            
+            # infer the dimensionality
+            self.dim = len(cells)
+
+            # set the evaluation type
+            self.eval_type = 'gridsample'
+
+            # keep track of the edge arrays
+            self.edgearrays = []
+
+            # keep track of the dimensions
+            self.edgedims = np.ones(self.dim,dtype='int')
+
+            # create the grid
+            for d in range(0,self.dim):
+                self.edgearrays.append(cells[d])
+                self.edgedims[d] = len(cells[d])
+
+        # 2. a list of tuples defining multiple arrays
+        # i.e. cells = [(np.linspace(-12,0,100),np.linspace(-4,0,50)),(np.linspace(0,12,100),np.linspace(0,4,50))]
+        # or a list of grids that have been preconstructed and stacked,
+        # i.e. [np.stack(np.meshgrid(np.linspace(-12,0,100),np.linspace(-4,0,50), indexing='ij'), axis=-1),np.stack(np.meshgrid(np.linspace(0,12,100),np.linspace(0,4,50), indexing='ij'), axis=-1)]
+        if isinstance(cells,list):
```

Comment:
Make `elif` to avoid catching 1D list (case 0)
```suggestion
        elif isinstance(cells,list):
```